nimble/controller/src/ble_ll_resolv.c

```diff
@@ -141,6 +156,14 @@ ble_ll_resolv_rpa_timer_cb(struct ble_npl_event *ev)
         ++rl;
     }
 
+#if MYNEWT_VAL(BLE_LL_HCI_VS_LOCAL_IRK)
+    OS_ENTER_CRITICAL(sr);
+    if (g_local_irk.is_set) {
+        generate_rpa(g_local_irk.irk, g_local_irk.rpa);
```

Comment:
RPA could be generated before critical section, and critical section could be only under if (as it may be cleared only from same task)